### ISSUE TITLE
Add titles to palette, use palette only

### ISSUE THREAD

**Issue by davidcasey** [Automattic/Iris#33]
I'd like the ability to add titles to the palette swatch (provided as a json array in palette: options)

Also I'd like the ability to use only the palette, removing the gradient color picker. This allows me to use the same tool for every color picking feature I want to implement.

**Comment by thirdender** [comment]
Not really. [CSS Specificity](http://www.stuffandnonsense.co.uk/archives/css_specificity_wars.html) allows you to set "more specific" CSS selectors which override the styles of "less specific" CSS selectors. The simplest example is something like:

    h1 { font-size: 1.5em; }
    h1.page-title { font-size: 1.8em; }

Because the second selector (`h1.page-title`) is more specific, those CSS properties will override the properties of the less specific selector (`h1`). This works regardless of where the two selectors are placed. But developers that don't know about specificity start by using `!important`, which is like using a sledgehammer to force your styles to take effect :-p That's when you should consider it a hack.

However, Javascript almost always applies styles using the `style` attribute on elements. This overrides almost all CSS stylesheet selectors, so you must use `!important` to override the Javascript assigned styles. The only downside is that now the Javascript style properties overridden in your `!important` CSS are not contolled by the JS, so you have to use this trick with care and cunning (specifically, if you override a JS set `width`, you'd better make sure your CSS `width: XX !important;` works with the JS's value gone/replaced).

**Comment by colinduwe** [comment]
Regardless of anyone's opinion regarding `!important` would you be open to PR to add a pallet only option? I'm looking here as a reference: https://bgrins.github.io/spectrum/#options-showPaletteOnly

**Comment by alfredvaa** [comment]
I would also like a "palette only" option.

**Comment by piersb** [comment]
Stick me down as a plus-one here.

**Comment by coreyworrell** [comment]
This is definitely useful, why would I **ever** want my client to choose ANY color?? What is this, 1995? Alright, aside from that little exaggeration, this would be incredibly useful, because honestly what is the use case for selecting a "visibly correct" color from the color wheel? Almost always you'll want to use an exact pre-selected color that you either enter into the text field, or select from a palette. This keeps your site's colors the same across the site.
+1

**Comment by abecuwedahive** [comment]
up
